homeassistant/components/nextcloud/config_flow.py

```diff
@@ -0,0 +1,169 @@
+"""Config flow for Nextcloud integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from nextcloudmonitor import NextcloudMonitor, NextcloudMonitorError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.event import track_time_interval
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("url"): str,
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+        vol.Optional("scan_interval", default=timedelta(seconds=60)): timedelta,
+        vol.Optional("verify_ssl", default=True): bool,
+    }
+)
+
+
+class NextCloud:
+    """Nextcloud class handler."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        url: str,
+        username: str,
+        password: str,
+        verify_ssl: bool,
+    ) -> None:
+        """Initialize."""
+        self.hass = hass
+        self.url = url
+        self.username = username
+        self.password = password
+        self.verify_ssl = verify_ssl
+        self.ncm = None
+
+    def setup(self):
+        """Call setup functions first to create connection with nextcloud server.
+
+        Raises:
+            NextcloudMonitorError: Connection was not established
+        """
+        self.ncm = NextcloudMonitor(
+            self.url, self.username, self.password, self.verify_ssl
+        )
+        self.__set_data_points_to_hass()
+
+    def update(self):
+        """Update download actual nextcloud data.
+
+        Raises:
+            UninitializedObject: Setup function was not called, or it failed
+        """
+        if self.ncm is None:
+            raise UninitializedObject()
+
+        self.ncm.update()
+        self.__set_data_points_to_hass()
+
+    def __set_data_points_to_hass(self):
+        self.hass.data[DOMAIN] = self.__get_ncm_data_points()
```

Comment:
I recommend using our data update coordinator to help fetch the data and keep a data cache.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities